sourcegraph/sourcegraph
# Web [Charts]: Add categorical bar chart component

Closes https://github.com/sourcegraph/sourcegraph/issues/32940
Closes https://github.com/sourcegraph/sourcegraph/issues/32942
Closes https://github.com/sourcegraph/sourcegraph/issues/32943

This PR adds a new chart component - the bar chart. This chart represents categorical-like data (it's different from series-like data where the x-axis acts as a timeframe, in categorical-like data x-axis is a set of different categories)

| Chart type  | Example |
| ------------- | ------------- |
| **Regular categories**  | <img width="428" alt="Screenshot 2022-06-13 at 23 45 48" src="https://user-images.githubusercontent.com/18492575/173393065-9f1af986-b06f-46d7-a815-481acf2ff14d.png">  |
| **Categories with sub groups**  | <img width="424" alt="Screenshot 2022-06-13 at 23 45 55" src="https://user-images.githubusercontent.com/18492575/173393210-c4db200e-f581-4781-a54b-b0a8ca58cd6b.png">  |
| **Categories with stacked groups**  | <img width="437" alt="Screenshot 2022-06-13 at 23 46 04" src="https://user-images.githubusercontent.com/18492575/173393293-82d87d98-1759-48a1-9132-fab2dd52cfe5.png">  |

This component also reuses the tooltip that we implemented during the line chart development. 

A few things to notice **that are not implemented/covered by this PR**
- This PR doesn't add a new bar chart for series-like (through time) data
- This PR's bar chart doesn't cover all corner cases around label positioning 

## Test plan
- Make sure that the chart has the right Y/X labels and correctly represents data sets in storybook stories
- Make sure that the tooltip works correctly in all modern browsers (Firefox, Safari, Chrome)



## App preview:

- [Web](https://sg-web-vk-create-bar-chart.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-zeuvocltfh.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
